tools/functional-tester/etcd-runner/main.go

```diff
@@ -212,6 +219,173 @@ func runRacer(eps []string, round int) {
 	doRounds(rcs, round)
 }
 
+func runWatcher(eps []string) {
+	runningTime := 60 * time.Second // time for which operation should be performed
+	noOfPrefixes := 20              // total number of prefixes which will be watched upon
+	watchPerPrefix := 10            // number of watchers per prefix
+	reqRate := 20                   // put request per second
+	keyPrePrefix := 1000            // max number of keyPrePrefixs for put operation
+
+	// pre generate prefixes equal to noOfPrefixes
+	prefixes := generateUniqueKeys(5, noOfPrefixes)
+	// pre generate keys
+	prePrefixKeys := generateRandomKeys(10, keyPrePrefix)
+	// map for holding prefixes and values
+	prefixKeysMap := make(map[string][]string)
+	// actual keys for performing put operation
+	keys := make([]string, 0)
+
+	// initialize the map with all prefixes
+	for _, prefix := range prefixes {
```

Comment:
@xiang90 If we have to keep it simple my suggestion would be, we start with watching on keys rather than on prefixes, that way the functionality of watching on key will be perfect as well as simple in every way and good for injecting defects.
What we will do then is rather than generating key on things we will have some pre-defined keys and we will randomly keep updating on these keys using randomly generated values.
That way code will be quite straight forward, simple and easier for error injections as you were suggesting.
What do you suggest/